cupy/core/core.pyx

```diff
@@ -2743,14 +2745,23 @@ not_equal = create_comparison(
 cpdef ndarray _convert_object_with_cuda_array_interface(a):
     cdef Py_ssize_t sh, st
     desc = a.__cuda_array_interface__
+    # TODO(leofang): enforce compliance to the latest protocol?
```

Comment:
I don't think we should enforce the protocol at the receiving side?

Reply:
OK I think I was wrong. NumPy didn't check the `version` field either: In `as_ctype()` they do:
https://github.com/numpy/numpy/blob/6bdbc1e2fd20df7272802df88f3790e5dd7ed057/numpy/ctypeslib.py#L532-L533
but in `PyArray_FromInterface()` (the actual workhorse) they don't:
https://github.com/numpy/numpy/blob/4d13d4cf8e8ab029f37e36753ca991c2d74c5cdb/numpy/core/src/multiarray/ctors.c#L2446
But, if we don't check the version, we need to be aware of backward-compatibility issues, which I think are handled correctly in the current PR.